Compilation errors with g++ 12

The code on develop branch fails to compile with g++ 12.x due to errors like:
```
[  0%] Building CXX object CMakeFiles/nextsim.dir/core/src/CommonRestartMetadata.cpp.o
In file included from /home/kornet/nextsimdg/core/src/include/ModelState.hpp:11,
                 from /home/kornet/nextsimdg/core/src/modules/include/IStructure.hpp:12,
                 from /home/kornet/nextsimdg/core/src/include/CommonRestartMetadata.hpp:11,
                 from /home/kornet/nextsimdg/core/src/CommonRestartMetadata.cpp:8:
/home/kornet/nextsimdg/core/src/include/ModelArray.hpp: In member function ‘double& Nextsim::ModelArray::operator[](size_t)’:
/home/kornet/nextsimdg/core/src/include/ModelArray.hpp:398:68: error: ‘as_const’ is not a member of ‘std’; did you mean ‘is_const’?
  398 |     double& operator[](size_t i) { return const_cast<double&>(std::as_const(*this)(i)); }
      |                                                                    ^~~~~~~~
      |                                                                    is_const
/home/kornet/nextsimdg/core/src/include/ModelArray.hpp: In member function ‘double& Nextsim::ModelArray::operator()(Args ...)’:
/home/kornet/nextsimdg/core/src/include/ModelArray.hpp:416:41: error: ‘as_const’ is not a member of ‘std’; did you mean ‘is_const’?
  416 |         return const_cast<double&>(std::as_const(*this)(args...));
      |                                         ^~~~~~~~
      |                                         is_const
make[2]: *** [CMakeFiles/nextsim.dir/build.make:202: CMakeFiles/nextsim.dir/core/src/CommonRestartMetadata.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:289: CMakeFiles/nextsim.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

--- a/core/src/include/ModelArray.hpp
+++ b/core/src/include/ModelArray.hpp
@@ -12,6 +12,7 @@
 #include <cstddef>
 #include <map>
 #include <string>
+#include <utility>
 #include <vector>
 
 namespace Nextsim {

